src/Features/CSharp/Portable/ExtractMethod/CSharpMethodExtractor.TriviaResult.cs

```diff
@@ -32,30 +32,49 @@ private CSharpTriviaResult(SemanticDocument document, ITriviaSavedResult result)
 
             protected override AnnotationResolver GetAnnotationResolver(SyntaxNode callsite, SyntaxNode method)
             {
-                if (callsite == null || !(method is MethodDeclarationSyntax methodDefinition))
+                if (callsite == null)
                 {
                     return null;
                 }
 
-                return (node, location, annotation) => AnnotationResolver(node, location, annotation, callsite, methodDefinition);
+                if (method is MethodDeclarationSyntax methodDefinition)
+                {
+                    return (node, location, annotation) => AnnotationResolver(node, location, annotation, callsite, methodDefinition.Body, methodDefinition.ExpressionBody, methodDefinition.SemicolonToken);
+                }
+                else if (method is LocalFunctionStatementSyntax localMethodDefinition)
```

Comment:
more unfortunate duplciation.  

I'm pausing on the review for now. IMO it needs to be restructured to keep all the logic uniform, and have only the subclasses have to do anything special about MethodDecls versus LocalFuncs.

Feel free to come to gitten if you'd like to discuss this at all. :)